app/include/zmk/endpoints.h

```diff
@@ -9,10 +9,7 @@
 #include <zmk/keys.h>
 #include <zmk/hid.h>
 
-enum zmk_endpoint {
-    ZMK_ENDPOINT_USB,
-    ZMK_ENDPOINT_BLE,
-};
+#include <zmk/events/endpoint_selection_changed.h>
 
 int zmk_endpoints_select(enum zmk_endpoint endpoint);
 int zmk_endpoints_toggle();
```

Comment:
Or in case you have some code that doesn't necessarily run on the endpoint changed event. Without being able to read the current endpoint directly from the source, you'd need to cache your own copy of it.

Reply:
You can! Further down in that same header is: `enum zmk_endpoint zmk_endpoints_selected();`.

Reply:
Oh right. I keep forgetting that GitHub shows context in a weird way that doesn't make it obvious when there's more of a file.